core/blockchain/impl/block_tree_impl.cpp

```diff
@@ -154,16 +156,17 @@ namespace kagome::blockchain {
         std::make_shared<TreeNode>(hash_res, header.number, nullptr, true);
     auto meta = std::make_shared<TreeMeta>(*tree);
 
-    BlockTreeImpl block_tree{std::move(header_repo),
-                             std::move(storage),
-                             std::move(tree),
-                             std::move(meta),
-                             std::move(extrinsic_observer),
-                             std::move(hasher),
-                             std::move(chain_events_engine),
-                             std::move(extrinsic_events_engine),
-                             std::move(runtime_core)};
-    return std::make_shared<BlockTreeImpl>(std::move(block_tree));
+    auto block_tree = new BlockTreeImpl(std::move(header_repo),
+                                        std::move(storage),
+                                        std::move(tree),
+                                        std::move(meta),
+                                        std::move(extrinsic_observer),
+                                        std::move(hasher),
+                                        std::move(chain_events_engine),
+                                        std::move(extrinsic_events_engine),
+                                        std::move(extrinsic_event_key_repo),
+                                        std::move(runtime_core));
+    return std::shared_ptr<BlockTreeImpl>(std::move(block_tree));
```

Comment:
Useless std::move. Pointers is trivial copy type.

Reply:
Fair enough.